api/src/main/java/uk/gov/dhsc/htbhf/claimant/message/MessageProcessor.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.dhsc.htbhf.claimant.message;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import org.springframework.util.CollectionUtils;
+import uk.gov.dhsc.htbhf.claimant.entity.Message;
+import uk.gov.dhsc.htbhf.claimant.repository.MessageRepository;
+
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.annotation.PostConstruct;
+
+/**
+ * Component that is triggered on a schedule and is responsible for finding all the messages that need to be
+ * processed and passing them off to the {@link MessageTypeProcessor} matching the {@link MessageType} of
+ * the message stored in the database.
+ */
+//TODO MRS 2019-04-23: Make this be constructed in a Configuration class, so construction of Map is in there, not here.
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class MessageProcessor {
+
+    private final List<MessageTypeProcessor> allMessageProcessors;
```

Comment:
Could just be `messageProcessors` and `messageProcessorsByType` (removing `all`)